### PR TITLE
CI: Check Patch Audit Compliance

### PR DESCRIPTION
**Pull request by reneme**
This adds an automated check to ensure that all referenced patches in the audit document are sufficiently denoted to assume that an audit was performed on it.

Note that I had to install `pyyaml` for `auditinfo`. Therefore, I needed to `poetry update` all downstream dependencies. Hence: sorry for the `poetry.lock` clutter in this PR.

Currently, this checks:

1. Is the patch authored, approved or audited by at least one authorative auditor, registered in config/auditors.yml
2. Is the patch classified regarding its relevance to the library's overall security

Note that we should not merge this until all currently registered patches are properly audited. Otherwise the `main` branch won't build anymore. As a side-effect this will also fail the Auto-Update pull request after the Bot created it. It will only turn green once an auditor clones it and performs an audit on the detected upstream patches.

### CODE REVIEW

**Comment by reneme** [comment]
For reference: the current result is this:

```
In topic 'Continuous Integration', the patch 'GH #3830 (92e5dd71e[25](https://github.com/sehlen-bsi/botan-docs/actions/runs/7182391225/job/19558798762?pr=167#step:9:26)f2121a22402e5ae590873d2ac0831)' is not sufficiently audited, because: Not classified
In topic 'Command Line Interface', the patch 'GH #3749 (73088f7b2ffdbf068e3022746b1437a9fab76a42)' is not sufficiently audited, because: Not classified
In topic 'Command Line Interface', the patch 'GH #3763 (5f502c695ab0cd38cab519e9da663e453fdf14ef)' is not sufficiently audited, because: Not classified
In topic 'Command Line Interface', the patch 'GH #3791 (84a828af77a98f2d03144a0ebb0c6d0cff383bd9)' is not sufficiently audited, because: Not classified
In topic 'Command Line Interface', the patch 'GH #3820 (6a6ba96a6a5d2e112b53c9c9b9cf66d62cdf5fef)' is not sufficiently audited, because: Not classified
In topic 'Documentation', the patch 'GH #3828 (79c8e235a6dce9d09df1cc605d283a4f735e0481)' is not sufficiently audited, because: Not classified
In topic 'PKI and Asymmetric Crypto', the patch 'GH #3770 (11b5d80790a65d6a7f8[26](https://github.com/sehlen-bsi/botan-docs/actions/runs/7182391225/job/19558798762?pr=167#step:9:27)7a95224f9e8c5ac0080)' is not sufficiently audited, because: Not classified
In topic 'PKI and Asymmetric Crypto', the patch 'GH #3786 (55b92d1af51591838[27](https://github.com/sehlen-bsi/botan-docs/actions/runs/7182391225/job/19558798762?pr=167#step:9:28)3ad[28](https://github.com/sehlen-bsi/botan-docs/actions/runs/7182391225/job/19558798762?pr=167#step:9:29)77b3b2ab2d1d51dc)' is not sufficiently audited, because: Not classified
In topic 'PKI and Asymmetric Crypto', the patch 'GH #3782 (7713fa936f3903f4d5ba2fd6b83702b4e3804d01)' is not sufficiently audited, because: Not classified
In topic 'PKI and Asymmetric Crypto', the patch 'GH #3784 (cef8f13347fe7f60b7e66519b0813e4c2c93d9af)' is not sufficiently audited, because: Not classified
In topic 'PKI and Asymmetric Crypto', the patch 'GH #3821 (b7c737ff816148fc05f634d2e3b25974bfd19452)' is not sufficiently audited, because: Not classified
In topic 'Release Preparation', the patch 'c2a2c0e6d8a1093493ec4931ffa09ba98ee66a12' is not sufficiently audited, because: No registered authorative auditor was involved in this patch
In topic 'TLS Improvements and Extensions', the patch 'GH #3764 (0981814435c005c02ebfdc312642c1414492efd1)' is not sufficiently audited, because: Not classified
In topic 'TLS Improvements and Extensions', the patch 'GH #3762 (565774637336a120ab818161cdb7755c87fbd6d8)' is not sufficiently audited, because: Not classified
In topic 'TLS Improvements and Extensions', the patch 'GH #3765 (fa143ad1484cca9b234b4e4ed41d4fa8d31e4366)' is not sufficiently audited, because: Not classified
In topic 'TLS Improvements and Extensions', the patch 'GH #3771 (20279c6b5115740b7987f9d5c6c[29](https://github.com/sehlen-bsi/botan-docs/actions/runs/7182391225/job/19558798762?pr=167#step:9:30)c[31](https://github.com/sehlen-bsi/botan-docs/actions/runs/7182391225/job/19558798762?pr=167#step:9:32)1d093bae)' is not sufficiently audited, because: Not classified
In topic 'TLS Improvements and Extensions', the patch 'GH #[38](https://github.com/sehlen-bsi/botan-docs/actions/runs/7182391225/job/19558798762?pr=167#step:9:39)20 (6a6ba96a6a5d2e112b53c9c9b9cf66d62cdf5fef)' is not sufficiently audited, because: Not classified
```

**Review by FAlbertDev — APPROVED** [review]
Nice! Exactly what I was looking for 👍

**Comment by reneme** [comment]
After merging all remaining audit pull requests, I rebased to `main` and will now wait for the new checks to turn green in CI.

**Pull request merged**
